pkg/lib/authenticationflow/declarative/intent_use_identity_login_id.go

```diff
@@ -52,6 +52,26 @@ func (n *IntentUseIdentityLoginID) CanReactTo(ctx context.Context, deps *authflo
 func (n *IntentUseIdentityLoginID) ReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows, input authflow.Input) (*authflow.Node, error) {
 	var inputTakeLoginID inputTakeLoginID
 	if authflow.AsInput(input, &inputTakeLoginID) {
+
+		flowRootObject, err := findFlowRootObjectInFlow(deps, flows)
+		if err != nil {
+			return nil, err
+		}
+
+		current, err := authflow.FlowObject(flowRootObject, n.JSONPointer)
+		if err != nil {
+			return nil, err
+		}
+		if currentBranch, ok := current.(config.AuthenticationFlowObjectBotProtectionConfigProvider); ok {
+			if IsBotProtectionRequired(currentBranch.GetBotProtectionConfig(), deps.Config.BotProtection) {
+				_, _, botProtectionVerified := authflow.FindMilestoneInCurrentFlow[MilestoneDidVerifyBotProtection](flows)
```

Comment:
It should read from the context to determine if bot protection is has done already.